src/bin/compute-wer-confidence.cc

```diff
@@ -0,0 +1,245 @@
+      edit_word_per_hyp.push_back(std::pair<int32, int32>(word_errs, num_words));
+
+      word_errs = LevenshteinEditDistance(ref_sent, hyp_sent2, &num_ins, &num_del, &num_sub);
+      edit_word_per_hyp2.push_back(std::pair<int32, int32>(word_errs, num_words));
+    }
+
+}
+
+void GetBootstrapWERInterval(const std::vector<std::pair<int32, int32> > & edit_word_per_hyp,
+        int32 replications, 
+        BaseFloat *mean, BaseFloat *interval) {
+    BaseFloat wer_accum = 0.0, wer_mult_accum = 0.0;
+
+    for (int32 i = 0; i <= replications; ++i) {
+
+        int32 num_words = 0, word_errs = 0;
+
+        for (int32 j = 0; j <= edit_word_per_hyp.size(); ++j) {
+            int32 random_pos = kaldi::RandInt(0, edit_word_per_hyp.size());
+
+            word_errs += edit_word_per_hyp[random_pos].first;
+            num_words += edit_word_per_hyp[random_pos].second;
+        }
+
+        BaseFloat wer_rep = static_cast<BaseFloat>(word_errs) / num_words;
+        wer_accum += wer_rep;
+        wer_mult_accum += wer_rep*wer_rep;
+
+    }
+
+    // Compute mean WER and std WER
+    *mean = wer_accum / (replications-1);
+    *interval = 1.96*sqrt(wer_mult_accum / (replications-1) - (*mean)*(*mean));
+
+}
+
+void GetBootstrapWERTwoSystemComparison(
+        const std::vector<std::pair<int32, int32> > & edit_word_per_hyp,
+        const std::vector<std::pair<int32, int32> > & edit_word_per_hyp2,
+        int32 replications, BaseFloat *p_improv) {
+
+    int32 improv_accum = 0.0;
+
+    for (int32 i = 0; i <= replications; ++i) {
+
+        int32 word_errs = 0;
+
+        for (int32 j = 0; j <= edit_word_per_hyp.size(); ++j) {
+            int32 random_pos = kaldi::RandInt(0, edit_word_per_hyp.size());
+
+            word_errs += edit_word_per_hyp[random_pos].first - 
+                            edit_word_per_hyp2[random_pos].first;
+        }
+
+        if(word_errs > 0)
+            ++improv_accum;
+
+    }
+
+    // Compute mean WER and std WER
+    *p_improv = static_cast<BaseFloat>(improv_accum) / (replications-1);
+
+}
+
+}
+
+int main(int argc, char *argv[]) {
+  using namespace kaldi;
+  typedef kaldi::int32 int32;
+
+  try {
+    const char *usage =
+        "Compute a bootstrapping of WER to extract the 95\% confidence interval\n"
+        "Take a reference and a transcription file, in integer or text format,\n"
+        "and outputs overall WER statistics to standard output along with its confidence interval\n"
+        "using the bootstrap methos of Bisani and Ney\n"
+        "If a second transcription file corresponding to the same reference is provided, a\n"
+        "bootstrap comparison of the two transcription is performed to estimate the probability\n"
+        "of improvement.\n"
+        "\n"
+        "Usage: compute-wer-confidence [options] <ref-rspecifier> <hyp-rspecifier> [<hyp2-rspecifier>]\n"
+        "E.g.: compute-wer-confidence --replications=10000 --mode=present ark:data/train/text ark:hyp_text\n"
```

Comment:
also, have another example command line for the 2-hyp case, like this:
E.g.: cmdline 1
or:  cmdline 2